src/Ryujinx.Ava/Input/AvaloniaMouseDriver.cs

```diff
@@ -63,16 +63,16 @@ private void Resized(Rect rect)
             _size = new Size((int)rect.Width, (int)rect.Height);
         }
 
-        private void Parent_ScrollEvent(object o, PointerWheelEventArgs args)
+        private void Parent_PointerWheelChanged(object o, PointerWheelEventArgs args)
         {
             Scroll = new Vector2((float)args.Delta.X, (float)args.Delta.Y);
         }
 
-        private void Parent_PointerReleaseEvent(object o, PointerReleasedEventArgs args)
+        private void Parent_PointerReleasedEvent(object o, PointerReleasedEventArgs args)
         {
             if (args.InitialPressMouseButton != Avalonia.Input.MouseButton.None)
             {
-                int button = (int)args.InitialPressMouseButton;
+                int button = (int)args.InitialPressMouseButton - 1;
 
                 if (PressedButtons.Count() >= button)
```

Comment:
```suggestion
                if ((uint)PressedButtons.Length > (uint)button)
```
- It is better to avoid using LINQ if we can, as it can be slow. In this case we can access the array length property directly.
- `>=` is not the right check since the button value being *equal* to the array length would be also out of range. It must be *less* than the array length.
- Casting the value to `uint` and doing a unsigned check also catches cases where `button` value is negative, so I believe that you can remove the `if (args.InitialPressMouseButton != Avalonia.Input.MouseButton.None)` check if you do that.

Reply:
This check is done in another place, it should be fixed there too.

Reply:
All fixed ;)